common/service/dynamicconfig/config.go

```diff
@@ -21,40 +21,54 @@
 package dynamicconfig
 
 import (
-	"fmt"
-	"sort"
+	"reflect"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 )
 
+const (
+	errCountLogThreshold = 1000
+)
+
 // NewCollection creates a new collection
 func NewCollection(client Client, logger log.Logger) *Collection {
-	return &Collection{client, logger, &sync.Map{}}
+	return &Collection{
+		client:   client,
+		logger:   logger,
+		keys:     &sync.Map{},
+		errCount: -1,
+	}
 }
 
 // Collection wraps dynamic config client with a closure so that across the code, the config values
 // can be directly accessed by calling the function without propagating the client everywhere in
 // code
 type Collection struct {
-	client Client
-	logger log.Logger
-	keys   *sync.Map
+	client   Client
+	logger   log.Logger
+	keys     *sync.Map // map of config Key to strongly typed value
+	errCount int64
```

Comment:
int or int32 should be enough?

Reply:
I can change this to int32 - but I am not sure if there is actually any benefit of doing so. Most modern processors are 64-bit and have 64 bit registers. For the same reason, int will be same as int64 on any modern server.